### ISSUE TITLE
Default null_value does not render correctly

### ISSUE THREAD

**Issue by kwinkunks** [abduhbm/zmapio#9]
Thank you for this library; I'm using it in [`gio`](https://code.agilescientific.com/gio/).

The default null value of `1e+30` is passed to a format string like `"{0:>{1}.{2}f}".format(1e30, 12, 4)` ([see line in code](https://github.com/abduhbm/zmapio/blob/main/zmapio/writer.py#L39)) and produces `'1000000000000000019884624838656.0000'` instead of `'1e+30` or possibly `'1.0000e+30'`.

Could fix by using the `str()` of the null value for NaNs, and the current format string for 'normal' values. Or you could change the format string to something like `"{0:>{1}.{2}e}"` (which would use scientific notation for *everything*, not sure if that's desirable).

I will submit a PR for a fix; you can decide what to do with it!

**Comment by baylejs** [comment]
Yes, been fighting this too.  I changed the format string to general "{0:>{1}.{2}g}" and tested the import of the resulting file. Value by value basis would be best as you also suggest.

**Comment by baylejs** [comment]
I had to spend a little more time on this for reading zmap file into a legacy mapping application. Got it working nicely now.

In zmap.py use a string for null.  null_value='0.1000000E+31',

In writer.py use conditional formatting depending on if the val is a number or a nan, which I set to -9999.999 temporarily for replacing with string null_value.

    fmt = "{0:>{1}.{2}f}".format
    width = zmap.field_width
    precision = zmap.decimal_places
    fmtNull = "{0:>{1}}".format
    # zmap.z_values = np.nan_to_num(zmap.z_values, nan=zmap.null_value)
    zmap.z_values = np.nan_to_num(zmap.z_values, nan=-9999.999)

    def write_lines(r):
        # line_gen = ("".join(line) + "\n" for line in chunks([fmt(val, width, precision) for val in r], nodes_per_line))
        line_gen = ("".join(line) + "\n" for line in chunks([fmt(val, width, precision) if val != -9999.999 else fmtNull(zmap.null_value, width) for val in r], nodes_per_line))
